Paws4Me/Paws4Me/Controllers/Logic/ViewController.swift

```diff
@@ -8,109 +8,227 @@
 import UIKit
 
 class ViewController: UIViewController, UISearchBarDelegate {
+    // MARK: - IBOulets
     @IBOutlet weak private var petTable: UITableView!
     @IBOutlet weak private var searchBar: UISearchBar!
-    private var filteredPetObject: AdoptPet?
-    private var adoptPetObject: AdoptPet?
-    override func viewDidLoad() {
-        super.viewDidLoad()
-        getAdoption()
-        petTable.delegate = self
-        petTable.dataSource = self
-        self.title = "Adoptable Animals"
-        setUpSearchbar()
-    }
+    @IBAction func btnCatTapped(_ sender: UIButton) {
+           petType = sender.titleLabel?.text?.lowercased() ?? ""
+           sender.layer.borderColor = UIColor.myAppPurple.cgColor
+           sender.layer.borderWidth = 2
+           searchPetType(type: petType)
+       }
+       @IBAction func btnKittenTapped(_ sender: UIButton) {
+           petType = sender.titleLabel?.text?.lowercased() ?? ""
+           sender.layer.borderColor = UIColor.myAppPurple.cgColor
+           sender.layer.borderWidth = 2
```

Comment:
I see what you are doing here 😄 something to maybe start thinking about is how can we extend the functionality of a UIButton so that we can sort of add this selected button look to all our buttons without duplicating the code? 

Maybe you can do some testing and then raise a separate PR where you improve this, but happy with it for now 😄

Reply:
I agree. I will definitely look into it For now I have just added the UIButton customisation to the UIButton extension